cmd/kubeadm/app/phases/uploadconfig/uploadconfig.go

```diff
@@ -59,4 +68,41 @@ func UploadConfiguration(cfg *kubeadmapi.MasterConfiguration, client clientset.I
 			kubeadmconstants.MasterConfigurationConfigMapKey: string(cfgYaml),
 		},
 	})
+	if err != nil {
+		return err
+	}
+
+	// Ensure that the BootstrapDiscoveryClusterRole exists
+	err = apiclient.CreateOrUpdateRole(client, &rbac.Role{
```

Comment:
I can't make up my mind whether this is better suited here or in the bootstraptoken phase.
Also, I'd like us to talk about whether it makes sense to do different types of tokens with different privileges for joining a master vs a node.

Reply:
I considered this as well, but currently in pkg `k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken` there are only things related to the cluster-info configMap.